tools/djxl_main.cc

```diff
@@ -550,57 +550,63 @@ int main(int argc, const char* argv[]) {
     if (args.print_read_bytes) {
       fprintf(stderr, "Decoded bytes: %" PRIuS "\n", decoded_bytes);
     }
-    if (args.alpha_blend) {
-      float background[3];
-      if (!ParseBackgroundColor(args.background_spec, background)) {
-        fprintf(stderr, "Invalid background color %s\n",
-                args.background_spec.c_str());
+    // When --disable_output was parsed, `filename_out` is empty and we don't
+    // need to write files.
+    if (!filename_out.empty()) {
+      if (args.alpha_blend) {
+        float background[3];
+        if (!ParseBackgroundColor(args.background_spec, background)) {
+          fprintf(stderr, "Invalid background color %s\n",
+                  args.background_spec.c_str());
+        }
+        AlphaBlend(&ppf, background);
       }
-      AlphaBlend(&ppf, background);
-    }
-    if (encoder) {
-      std::ostringstream os;
-      os << args.jpeg_quality;
-      encoder->SetOption("q", os.str());
-    }
-    if (encoder && args.use_sjpeg) {
-      encoder->SetOption("jpeg_encoder", "sjpeg");
-    }
-    jxl::extras::EncodedImage encoded_image;
-    if (encoder) {
-      if (!args.quiet) cmdline.VerbosePrintf(2, "Encoding decoded image\n");
-      if (!encoder->Encode(ppf, &encoded_image)) {
-        fprintf(stderr, "Encode failed\n");
-        return EXIT_FAILURE;
+      if (encoder) {
```

Comment:
Now we can get rid of all the if (encoder ... parts, because we would have exited early in line 532 otherwise

Reply:
done, and I used `encoder` instead of `!filename_out.empty()` in the main if, they should be the same value at that point.